hooks/pre/35-change-evr-extension-ownership.rb

```diff
@@ -0,0 +1,26 @@
+# In Katello 4.16, the 'evr' extension is removed from PostgreSQL and integrated into the Katello database via a migration.
+# This hook ensures the 'evr' extension's ownership is transferred to the 'foreman' user so migrations can act on it.
+
+if local_postgresql? && execute("rpm -q postgresql-evr", false, false)
+  is_postgresql_active = execute_command("systemctl is-active postgresql", false, true)&.first&.strip == "active"
+
+  # Ensure the PostgreSQL service is running
+  unless is_postgresql_active
+    logger.debug("Starting postgresql service")
+    start_services(['postgresql']) unless app_value(:noop)
+  end
+
+  # Update the ownership of the evr extension
+  logger.debug("Updating ownership of the evr extension if it is enabled")
+  database = param_value('foreman', 'db_database') || 'foreman'
+  username = param_value('foreman', 'db_username') || 'foreman'
+  sql = "psql -d '#{database}' -c \\\"UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='#{username}') WHERE extname='evr';\\\""
```

Comment:
Can you re-factor the way the SQL is handled to re-use the functions added via https://github.com/theforeman/foreman-installer/pull/984/files to ensure no passwords are leaked?

Reply:
This one is a bit different in that we run it as `postgres` (because we assume that's a superuser), relying on ident auth so no actual passwords are used. Since we don't consider the DB name and username to be that secret, it probably is ok.